pkg/cluster/kubernetes/sync.go

```diff
@@ -377,6 +377,11 @@ func applyMetadata(res resource.Resource, syncSetName, checksum string) ([]byte,
 		mixin["labels"] = mixinLabels
 	}
 
+	namespace, _, _ := res.ResourceID().Components()
+	if namespace != kresource.ClusterScope {
+		mixin["namespace"] = namespace
```

Comment:
Nice and simple. Much better than my suggestion at https://github.com/fluxcd/flux/issues/2797#issuecomment-579900826

Can you please add a detailed comment explaining why we are doing this?

Reply:
done

Reply:
Fantastic